build.rs

```diff
@@ -1,3 +1,3 @@
 fn main() {
-    println!("cargo:rerun-if-changed=progs/")
+    println!("rerun-if-changed=progs/")
```

Comment:
This _introduces_ the bug for me. With this change (i.e. without `cargo:`) the tests are recompiled if i `touch` README.md, but with `cargo:` intact it works like expected.

Reply:
https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-changed If no `cargo:rerun-if-changed` is emitted then the project is always recompiled.

Have you tested that this change doesn't recompile if you only touch some unrelated file?

Reply:
With `cargo:rerun-if-changed` changing and adding tests doesn't trigger a recompile. And I've wasted enough time writing tests that don't work.

I'd rather have false positives than false negatives.

Reply:
Also:
https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-changed
Says:
> If the path points to a directory, it will scan the entire directory for any modifications.

It might only go one level down. So:
 - A
     - B
         - D
     - C
 My problem is I change D and a recompile is not issued and it caches the old tests in the build file, giving me the same result. And that might be because it's only caching B and C. So we might have to change our build-script to look more like the test-finding thing.
 
 I understand your frustration, and I agree that it's annoying to always rebuild everything when all files are touched. But the problems of having the tests you write pass locally and fail on the CI is more annoying since it's invicible.